### DISCUSSION TITLE
Issue while saving file containing foreign language letters in its name

### DISCUSSION THREAD

**Discussion by SeenaThite** [ImageMagick/ImageMagick#5561] [Help]
Hi,

I am trying to save a jpeg file using Imagemagick api in a folder whose name has some alphabets other than english (e.g. Ready-Napetí-Napetí1 ) But the write function is throwing exception and file is not getting saved. 

My current code is

	Magick::Image image;

	try
	{
		image.read(filePath);
		image.formatExpression(format); 
		image.write(originalPath);	
	}
Here image.write function throws exception if my path contains some non english alphabets.
Do I need to any different setting for this or use some different API?

 I am using Imagemagick windows version 7. 

Thanks,
Seena

**Comment by snibgo** [comment]
What version of IM?

> Here image.write function throws exception ...

What exception is thrown?

Using your supplied filename, it works fine for me, with IM v7.1.0-20 on Windows 8.1.

**Reply by SeenaThite** [reply]
IM version 7.1.0 for windows. 
Can you try to create a folder named Ready-Napetí-Napetí1 and save a jpg file in that folder. It doesn't work.
The exception I receive is "Unknown exception"

**Comment by snibgo** [comment]
Are you catching exceptions?

I have no problems:

```
f:\web\im>md Ready-Napetí-Napetí1

f:\web\im>dir Ready-Napetí-Napetí1
 Volume in drive F is New Volume
 Volume Serial Number is F8C5-053E

 Directory of f:\web\im\Ready-Napetí-Napetí1

13/09/2022  15:46    <DIR>          .
13/09/2022  15:46    <DIR>          ..
               0 File(s)              0 bytes
               2 Dir(s)  132,126,670,848 bytes free

f:\web\im>%IM7DEV%rdwr2 rose: Ready-Napetí-Napetí1\x.png
8
f:\web\im>dir Ready-Napetí-Napetí1
 Volume in drive F is New Volume
 Volume Serial Number is F8C5-053E

 Directory of f:\web\im\Ready-Napetí-Napetí1

13/09/2022  15:46    <DIR>          .
13/09/2022  15:46    <DIR>          ..
13/09/2022  15:46             6,994 x.png
               1 File(s)          6,994 bytes
               2 Dir(s)  132,126,662,656 bytes free
```
My program, rdwr2.cpp, is:
```
#include <Magick++.h>
#include <string>
#include <iostream>

using namespace std;

using namespace Magick;

int main( int /*argc*/, char ** argv)
{
  InitializeMagick(*argv);

  Image image;

  try 
  { 
    image.read(argv[1]);

    cout << image.depth();

    image.write(argv[2]);

  } 

  catch( exception &error_ )
    {
      cout << "Caught exception: " << error_.what() << endl;
      return 1;
    }

  return 0;
}
```

**Comment by SeenaThite** [comment]
Yes I am catching the exception. The error message is 
"WriteBlob Failed `C:\bugs\1210430\4257437-ZPD-000-SW2022\Ready-Napetí-Napetí1\4257437-ZPD-000.PNG' @ error/png.c/MagickPNGErrorHandler/1715"

I am reading a tiff file and saving it as png/jpeg

**Comment by snibgo** [comment]
What is your input file? What is your complete program?

EDIT: Are you sure that directory exists?

**Comment by SeenaThite** [comment]
bool  Save(CString& path, CString& extn)
{
	InitializeMagick("");

	std::string filePath(CW2A(path.GetString()));   //This is tiff file path
	std::string format(CW2A(extn.GetString()));    //This is format (png/jpg)
	std::string originalPath(CW2A(m_OriginalPSDPath.GetString()));  //This is the output png path

	Magick::Image image;

	try
	{
		image.read(filePath);   // Reading a tiff file here
		image.formatExpression(format); //setting png format here
		image.write(originalPath);	//passing the output png path here which does exist
	}
	catch (Exception& error)
	{
		auto err = error.what();
		
		::DeleteFile(GetTempTifPath());
	}

	return true;
}

And yes the output directory does exist. I have manually created the output folder.

**Comment by SeenaThite** [comment]
I checked further and found that i am getting a nested exception. So along with exception "WriteBlob Failed `C:\bugs\1210430\4257437-ZPD-000-SW2022\Ready-Napetí-Napetí1\4257437-ZPD-000.PNG' @ error/png.c/MagickPNGErrorHandler/1715"
I am also getting this exception
"UnableToOpenBlob 'C:\bugs\1210430\4257437-ZPD-000-SW2022\Ready-Napetí-Napetí1\4257437-ZPD-000.PNG': No such file or directory @ error/blob.c/OpenBlob/3527"

Do I need to do any unicode settings or something else to support characters of all languages?

**Comment by snibgo** [comment]
Perhaps the problem is in your use of "`CW2A`". This assumes its input is `LPWSTR` (wide char) and that your current code page is UTF-8. Either of these assumptions may be false.
